app/models/spree/order.rb

```diff
@@ -713,15 +713,15 @@ def associate_customer
     def ensure_customer
       return if associate_customer
 
-      self.customer = Customer.new(
+      self.customer = Customer.create(
         enterprise: distributor,
         email: email_for_customer,
         user: user,
-        name: bill_address&.full_name,
+        first_name: bill_address&.first_name.to_s,
+        last_name: bill_address&.last_name.to_s,
         bill_address: bill_address&.clone,
         ship_address: ship_address&.clone
       )
-      customer.save
 
       Bugsnag.notify(customer.errors.full_messages.join(", ")) unless customer.persisted?
     end
```

Comment:
I don't think that you increased the complexity of this method much but if we want to satisfy Rubocop we can write this method as `associate_customer || create_customer`. Remove the Bugsnag notification and move the customer creation into its own method. That's much easier to read, actually.

Reply:
Campsite rule: Leave the campsite better than you found it.